pkg/grpc/requestinfo/headers_multimap.go

```diff
@@ -7,22 +7,39 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
-// HeadersMultiMap is an interface for http.Header and for metadata.MD types.
-// The former will require a wrapper to implement this interface.
-type HeadersMultiMap interface {
+// HeadersMultiMapR is an interface for http.Header and for metadata.MD types.
```

Comment:
I'm also wondering what do we gain by distinguishing between reader and writer interface? It feels unnecessarily abstract and Java-like to me. I'd suggest to keep the single `HeadersMultiMap` interface and just add a `Set`.

Reply:
I think there were some benefits, but there are none since refactoring.